test/test_standard_bookmark.py

```diff
@@ -0,0 +1,196 @@
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_key_value)
+                    self.assertIsNotNone(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    # assumes no changes to data during test
+                    if not stream == "users":
+                        self.assertEqual(second_bookmark_value,
+                                        first_bookmark_value)
+
+                    for record in first_sync_messages:
+
+                        # Verify the first sync bookmark value is the max replication key value for a given stream
+                        replication_key_value = record.get(replication_key)
+                        if stream == "tickets":
+                            replication_key_value = datetime.utcfromtimestamp(replication_key_value).strftime('%Y-%m-%dT%H:%M:%SZ')
+                      
+                        self.assertLessEqual(
+                            replication_key_value, first_bookmark_value_utc,
+                            msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+                    
+                    for record in second_sync_messages:
+                        # Verify the second sync replication key value is Greater or Equal to the first sync bookmark
+                        replication_key_value = record.get(replication_key)
+
+                        if stream == "tickets":
+                            replication_key_value = datetime.utcfromtimestamp(replication_key_value).strftime('%Y-%m-%dT%H:%M:%SZ')
+                            
+                        if stream in ["group_memberships", "ticket_fields", "macros"]:
+                            replication_key_value = datetime.strptime(replication_key_value, '%Y-%m-%dT%H:%M:%S.000000Z').strftime('%Y-%m-%dT%H:%M:%SZ')
+                        
+                        self.assertGreaterEqual(replication_key_value, simulated_bookmark_value,
+                                                msg="Second sync records do not repect the previous bookmark.")
+
+                        # Verify the second sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(
+                            replication_key_value, second_bookmark_value_utc,
+                            msg="Second sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+
+                    # verify that you get less data the 2nd time around
+                    self.assertLess(
+                        second_sync_count,
+                        first_sync_count,
+                        msg="second sync didn't have less records, bookmark usage not verified")
```

Comment:
This assertion is not totally necessary here, as the test is checking by replication-key values that bookmarking is working as expected. It can be removed.

Reply:
removed it